### ISSUE TITLE
Event when routing is cancelled working only once

### ISSUE THREAD

**Issue by e-altintas** [NaluKit/nalu#315]
When the user cancel the routing, an event ROUTING_CANCELLED_BY_USER is triggered. If the user repeat the same action, event is not triggered anymore

**Comment by FrankHossfeld** [comment]
Thanks for reporting. 

I have not seen this behavior before. Usually a routing starts and the START_ROUTING event is fired. Depending on what happens, we will see one of the following events:

* ROUTING_DONE: everything is fine
* ROUTING_CANCELLED_BY_USER routing is canceled by the user
* ROUTING_ABORTED f.e. by a a RoutingInterceptionException

So, to see the ROUTING_CANCELLED_BY_USER again, you need to start a new routing.

Please can you provide more information. 

* What version of Nalu are you using? 
* What is your use case to expect the ROUTING_CANCELLED_BY_USER event twice?

**Comment by FrankHossfeld** [comment]
Thanks for the added information. I will look into it.

**Comment by FrankHossfeld** [comment]
I'll updated the [https://github.com/NaluKit/nalu-examples](https://github.com/NaluKit/nalu-examples) and add a handler to the nalu-complex-app-example that log RouteStateEvent values. The events that get logged looks as expected.

You can try it by yourself. just select a person change something and navigate ...

  You might look why we see the ROUTING_ABORTED state. Usually this means, something is wrong during a routing occurs. I suggest to debug the `RouterStateEventFactory` and set a breaking inside the `fireAbortRoutingEvent`-method. That might give you an idea why this event is fired.
  
  By the way, you can just move to Nalu 2.12.1. There are no breaking changes.

**Comment by e-altintas** [comment]
Hello,

Thanks for the response. I managed to compile the nalu-complex-app-example but there are no data, I don't know how to add data for my testing
![image](https://github.com/NaluKit/nalu/assets/124250801/c5860d77-8e4e-4454-8e6a-e72e6da96fea)

Best regards,

Ervin

**Comment by FrankHossfeld** [comment]
Oh sorry, forgot to mention, search for 'Duck' or 'Simpsons'.

**Comment by FrankHossfeld** [comment]
@e-altintas any update on this issue?

**Comment by e-altintas** [comment]
Hello, sorry I didn't had time to test it yet. I will let you know

**Comment by e-altintas** [comment]
Thank you, I managed to fix my bug by clearing the router state when the user cancel the routing.

![image](https://github.com/NaluKit/nalu/assets/124250801/b2a60dff-45d7-480c-80e8-75288132cdfe)

**Comment by FrankHossfeld** [comment]
Hmmm, I am a little bit surprised. Usually, there is no need to cancelRoutiing() or clear the factory. There might be something in your code, that prevent  the router from doing it.

**Comment by FrankHossfeld** [comment]
I'll checked the code and I am pretty sure, that everything is fine.

The `RouterStateEventFactory` is firing the `RouterStateEvent`-event with the value **ROUTING_CANCELED_BY_USER** state and just right after that, the `RouterStateEvent`-event is fired again with the value **ROUTING_DONE** and the RouterStateEventFactory is cleared!

The only way - that I currently can imagine - to avoid this is, that the `RouterStateEvent`-event is caught and in case the value  is **ROUTING_CANCELED_BY_USER** the application fires an exception and the normal program flow is intercepted.

What does the `cancelRouting`-method do?
Is the code throwing an exception in case a `RouterStateEvent`-event is caught with the value **ROUTING_CANCELED_BY_USER**?

That is only situation, i can imagine, why this happens.

**Comment by FrankHossfeld** [comment]
Hi Ervin,
thanks for responding. I'll close the issue. If you ran into trouble again, you can create a new issue or reopen this one .
For general questions: [Nalu Gitter Room](https://matrix.to/#/#Nalukit42_Lobby:gitter.im)
cu Frank
